[CI] LogicalPlanOptimizerTests testPlanSanityCheckWithBinaryPlans failing

**Build Scans:**
- [elasticsearch-intake #14453 / part3](https://gradle-enterprise.elastic.co/s/455dshmmv2ci6)
- [elasticsearch-intake #14454 / part3](https://gradle-enterprise.elastic.co/s/jsngy5kfr4ohk)
- [elasticsearch-pull-request #46446 / part-3-arm](https://gradle-enterprise.elastic.co/s/d6dkptqhmrack)
- [elasticsearch-pull-request #46446 / part-3](https://gradle-enterprise.elastic.co/s/pusy7epuxyzny)
- [elasticsearch-pull-request #46450 / part-3](https://gradle-enterprise.elastic.co/s/a6pstilcg3yvy)
- [elasticsearch-pull-request #46445 / part-3](https://gradle-enterprise.elastic.co/s/zcbsnlsbmcxii)
- [elasticsearch-pull-request #46435 / part-3](https://gradle-enterprise.elastic.co/s/a62webth7zsce)
- [elasticsearch-intake #14450 / part3](https://gradle-enterprise.elastic.co/s/7yegnf7b63q4u)
- [elasticsearch-pull-request #46425 / part-3](https://gradle-enterprise.elastic.co/s/tixhcvkiy47s4)
- [elasticsearch-pull-request #46412 / part-3](https://gradle-enterprise.elastic.co/s/6t6uv4ihq6oco)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.optimizer.LogicalPlanOptimizerTests.testPlanSanityCheckWithBinaryPlans" -Dtests.seed=55518B61A88DF8BE -Dtests.locale=szl -Dtests.timezone=America/Detroit -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.optimizer.LogicalPlanOptimizerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testPlanSanityCheckWithBinaryPlans),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an instance of org.elasticsearch.xpack.esql.plan.logical.Limit
     but: <Join[LEFT,[language_code{r}#441285],[language_code{r}#441285],[language_code{f}#441298]]
|_EsqlProject[[_meta_field{f}#441293, emp_no{f}#441287, first_name{f}#441288, gender{f}#441289, hire_date{f}#441294, job{f}#
441295, job.raw{f}#441296, languages{f}#441290 AS language_code, last_name{f}#441291, long_noidx{f}#441297, salary{f}#441292]]
| \_Limit[1000[INTEGER]]
|   \_EsRelation[test][_meta_field{f}#441293, emp_no{f}#441287, first_name..]
\_EsRelation[languages_lookup][LOOKUP][language_code{f}#441298, language_name{f}#441299]> is a org.elasticsearch.xpack.esql.plan.logical.join.Join
```

**Issue Reasons:**
- [8.x] 11 consecutive failures in test testPlanSanityCheckWithBinaryPlans
- [8.x] 4 consecutive failures in step part3
- [8.x] 6 consecutive failures in step part-3
- [8.x] 4 consecutive failures in pipeline elasticsearch-intake
- [8.x] 6 consecutive failures in pipeline elasticsearch-pull-request
- [8.x] 11 failures in test testPlanSanityCheckWithBinaryPlans (91.7% fail rate in 12 executions)
- [8.x] 4 failures in step part3 (100.0% fail rate in 4 executions)
- [8.x] 6 failures in step part-3 (85.7% fail rate in 7 executions)
- [8.x] 4 failures in pipeline elasticsearch-intake (100.0% fail rate in 4 executions)
- [8.x] 6 failures in pipeline elasticsearch-pull-request (85.7% fail rate in 7 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -317,9 +317,6 @@ tests:
 - class: org.elasticsearch.reservedstate.service.FileSettingsServiceTests
   method: testInvalidJSON
   issue: https://github.com/elastic/elasticsearch/issues/116521
-- class: org.elasticsearch.xpack.esql.optimizer.LogicalPlanOptimizerTests
-  method: testPlanSanityCheckWithBinaryPlans
-  issue: https://github.com/elastic/elasticsearch/issues/118656
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java
@@ -4911,8 +4911,7 @@ public class LogicalPlanOptimizerTests extends ESTestCase {
             """);
 
         var project = as(plan, Project.class);
-        var limit = as(project.child(), Limit.class);
-        var join = as(limit.child(), Join.class);
+        var join = as(project.child(), Join.class);
 
         var joinWithInvalidLeftPlan = join.replaceChildren(join.right(), join.right());
         IllegalStateException e = expectThrows(IllegalStateException.class, () -> logicalOptimizer.optimize(joinWithInvalidLeftPlan));

